geode-core/src/main/java/org/apache/geode/internal/tcp/Connection.java

```diff
@@ -1502,25 +1503,15 @@ public void run() {
         }
       }
 
-      releaseInputBuffer();
-
       // make sure that if the reader thread exits we notify a thread waiting for the handshake.
       notifyHandshakeWaiter(false);
       readerThread.setName("unused p2p reader");
       synchronized (stateLock) {
         isRunning = false;
         readerThread = null;
       }
-    }
-  }
 
-  private void releaseInputBuffer() {
-    synchronized (inputBufferLock) {
-      ByteBuffer tmp = inputBuffer;
-      if (tmp != null) {
-        inputBuffer = null;
-        getBufferPool().releaseReceiveBuffer(tmp);
-      }
+      inputSharing.close();
```

Comment:
I'm wondering if this method needs to claim its own reference to the buffer to balance this close()